src/integer/z/arithmetic/add.rs

```diff
@@ -398,3 +447,68 @@ mod test_add_between_z_and_zq {
         assert_eq!(e, Zq::from((0, i64::MAX)));
     }
 }
+
+#[cfg(test)]
+mod test_add_between_z_and_poly_over_z {
+    use super::Z;
+    use crate::integer::PolyOverZ;
+    use std::str::FromStr;
+
+    /// Testing addition for [`Z`] and [`PolyOverZ`]
+    #[test]
+    fn add() {
+        let a: Z = Z::from(9);
+        let b: PolyOverZ = PolyOverZ::from_str("4  1 1 2 3").unwrap();
+        let c: PolyOverZ = a + b;
+        assert_eq!(c, PolyOverZ::from_str("4  10 1 2 3").unwrap());
+    }
+
+    /// Testing addition for both borrowed [`Z`] and [`PolyOverZ`]
+    #[test]
+    fn add_borrow() {
+        let a: Z = Z::from(9);
+        let b: PolyOverZ = PolyOverZ::from_str("4  1 1 2 3").unwrap();
+        let c: PolyOverZ = &a + &b;
+        assert_eq!(c, PolyOverZ::from_str("4  10 1 2 3").unwrap());
+    }
+
+    /// Testing addition for borrowed [`Z`] and [`PolyOverZ`]
+    #[test]
+    fn add_first_borrowed() {
+        let a: Z = Z::from(9);
+        let b: PolyOverZ = PolyOverZ::from_str("4  1 1 2 3").unwrap();
+        let c: PolyOverZ = &a + b;
+        assert_eq!(c, PolyOverZ::from_str("4  10 1 2 3").unwrap());
+    }
+
+    /// Testing addition for [`Z`] and borrowed [`PolyOverZ`]
+    #[test]
+    fn add_second_borrowed() {
+        let a: Z = Z::from(9);
+        let b: PolyOverZ = PolyOverZ::from_str("4  1 1 2 3").unwrap();
+        let c: PolyOverZ = a + &b;
+        assert_eq!(c, PolyOverZ::from_str("4  10 1 2 3").unwrap());
+    }
```

Comment:
possibly batch them for brevity and call the test itself availability as we have done for some other types

@Marcel583 can probably look for what is most consistent, but I would opt for a less repetititve design (at least right now I would)

Reply:
I have talked to @Marcel583 who said this is the consistent solution for now, but we should have a look at all test in a refactoring